pkg/admission/validator/shoot.go

```diff
@@ -179,6 +179,9 @@ func (s *shoot) validateUpdate(oldShoot, shoot *core.Shoot, cloudProfile *garden
 
 	allErrs = append(allErrs, s.validateShoot(shoot, oldInfraConfig, infraConfig, cloudProfile, cpConfig)...)
 
+	// TODO(dkistner) remove this once csi-driver async operation issues are resolved.
+	allErrs = append(allErrs, azurevalidation.ValidateUpgradeV120ToV121(shoot, oldShoot, specPath.Child("kubernetes", "version"))...)
```

Comment:
Probably it also makes sense to make this validation controlled by a flag/configuration option (enabled by default). This can be helpful for Gardener operators, they will be able to disable this new validation, if they need. WDYT?

Reply:
What would be the deprecation policy after we update to a CSI driver that fixes the issue ? 

Something like this ?
```
+1 version -> remove check internally
+2 versions -> remove flag
```

Reply:
I don't have strong opinion. Yes, we can keep the validation in-place (disabled by default) until we gain experience with `azuredisk-csi-driver@v1.16.0`. Anyways, we can decide later how/when to proceed with removal of this validation/restriction.